osm2streets/src/transform/collapse_intersections.rs

```diff
@@ -188,7 +191,7 @@ pub fn collapse_intersection(streets: &mut StreetNetwork, i: NodeID) {
     // Simplify curves and dedupe points. The epsilon was tuned for only one location that was
     // breaking
     let epsilon = 1.0;
-    new_road.osm_center_points = Pt2D::simplify_rdp(new_road.osm_center_points, epsilon);
+    new_road.osm_center_points = PolyLine::must_new(Pt2D::simplify_rdp(new_pts, epsilon));
```

Comment:
The simplify_rdp will dedupe adjacent points. The only way this could crash is if the entire polyline doubles back on itself, which shouldn't happen for these cases where we're removing a short road